### ISSUE TITLE
"not" filters don't filter UAs picked with per-country statistics

### ISSUE THREAD

**Issue by Kovensky** [ai/browserslist#125]
node session with browserslist@1.7.4:

```js
> require('browserslist')(['> 1%'])
[ 'and_chr 56',
  'and_uc 11',
  'android 4.4',
  'chrome 55',
  'chrome 49',
  'edge 14',
  'firefox 50',
  'ie 11',
  'ios_saf 10.0-10.2',
  'op_mini all',
  'safari 10',
  'samsung 4' ]
> require('browserslist')(['> 1%', 'not and_chr 56'])
[ 'and_uc 11', // filtered out here
  'android 4.4',
  'chrome 55',
  'chrome 49',
  'edge 14',
  'firefox 50',
  'ie 11',
  'ios_saf 10.0-10.2',
  'op_mini all',
  'safari 10',
  'samsung 4' ]
> require('browserslist')(['> 1% in JP', 'not and_chr 56'])
[ 'and_chr 56', // not filtered out!
  'chrome 55',
  'chrome 49',
  'edge 14',
  'firefox 51',
  'firefox 50',
  'ie 11',
  'ios_saf 10.0-10.2',
  'ios_saf 9.3',
  'safari 10' ]
```

**Comment by ai** [comment]
Done 134573b

Thanks for report.

**Comment by ai** [comment]
Released in 1.7.5.
